File: InsertionSort.js
function InsertionSort(arr)
{
  for(let i=1;i<arr.length;i++)
  {
    let ele=arr[i];
    let j=i-1;    // sorted resion
    while(j>=0 && ele<arr[j])
    {
        arr[j+1]=arr[j];
        j--;
    }
    arr[j+1]=ele;
  }
  return arr;
}
let arr=[5,4,3,2,1];
console.log(InsertionSort(arr));

File: Test3/InsertionSort.java
package Test3;

public class InsertionSort {
    public static void main(String[] args) {
        int arr[]={1,3,4,5,2};
        for(int i=1;i<arr.length;i++){
            InsertLast(arr,i);
        }
        for(int i=0;i<arr.length;i++){
            System.out.print(arr[i]+" ");
        }
    }
    public static void InsertLast(int arr[],int i){
        int item=arr[i];
        int j=i-1;
        while(j>=0 && arr[j]>item){
            arr[j+1]=arr[j];
            j--;
        }
        arr[j+1]=item;
    }
}
